shared/src/commonMain/kotlin/org/hyperskill/app/step/presentation/StepReducer.kt

```diff
@@ -1,33 +1,78 @@
 package org.hyperskill.app.step.presentation
 
+import org.hyperskill.app.step.domain.analytic.StepViewedHyperskillAnalyticEvent
+import org.hyperskill.app.step.domain.model.StepRoute
 import org.hyperskill.app.step.presentation.StepFeature.Action
 import org.hyperskill.app.step.presentation.StepFeature.Message
 import org.hyperskill.app.step.presentation.StepFeature.State
-import org.hyperskill.app.step.domain.analytic.StepViewedHyperskillAnalyticEvent
+import org.hyperskill.app.step_completion.presentation.StepCompletionFeature
+import org.hyperskill.app.step_completion.presentation.StepCompletionReducer
 import ru.nobird.app.presentation.redux.reducer.StateReducer
 
-class StepReducer : StateReducer<State, Message, Action> {
+class StepReducer(private val stepRoute: StepRoute, private val stepCompletionReducer: StepCompletionReducer) : StateReducer<State, Message, Action> {
     override fun reduce(state: State, message: Message): Pair<State, Set<Action>> =
         when (message) {
             is Message.Initialize ->
                 if (state is State.Idle ||
                     (message.forceUpdate && (state is State.Data || state is State.Error))
                 ) {
-                    State.Loading to setOf(Action.FetchStep(message.stepRoute))
+                    State.Loading to setOf(Action.FetchStep(stepRoute))
                 } else {
                     null
                 }
-            is Message.StepLoaded.Success ->
-                State.Data(message.step) to emptySet()
+            is Message.StepLoaded.Success -> {
+                State.Data(
+                    step = message.step,
+                    isPracticingAvailable = stepRoute is StepRoute.Learn,
+                    stepCompletionState = StepCompletionFeature.State(
+                        currentStep = message.step,
+                        continueButtonAction = if (stepRoute is StepRoute.Learn) {
+                            StepCompletionFeature.ContinueButtonAction.FetchNextStepQuiz
+                        } else {
+                            StepCompletionFeature.ContinueButtonAction.NavigateToBack
+                        }
+                    )
```

Comment:
In my opinion, it's better to move `StepCompletionFeature.State` creation to `StepCompletionFeature` static method. I think `StepCompletionFeature` should decide how to create its own state depending on conditions.